### PR TITLE
docs: add note about ksqlDB headless mode and Avro schemas (DOCS-1487)

### PR DESCRIPTION
**Pull request by JimGalasyn**

### CODE REVIEW

**Comment by JimGalasyn** [comment]
@londoncalling Sadly, there's no easy way. I copy the changed files manually into my clone of the main docs repo and build locally with `make html`, but this is tedious.

**Comment by JimGalasyn** [comment]
@big-andy-coates, I reworked the SR section, how does it look now?

File: docs/installation/server-config/index.rst [hunk 1]
```diff
@@ -236,6 +236,10 @@ server is running a predefined script, it will automatically disable its REST en
 
 .. tip:: When both the ``ksql.queries.file`` property and the ``--queries-file`` argument are present, the ``--queries-file`` argument will take precedence.
 
+When you run a KSQL application that uses Avro in interactive mode, KSQL
+infers schemas from |sr| automatically, but in non-interactive mode,
+you must provide schemas manually.
```
Comment (big-andy-coates) [inline docs/installation/server-config/index.rst]: Out of whack with the change above:  non-interactive does talk to SR.

File: docs-md/operate-and-deploy/installation/server-config/index.md [hunk 2]
```diff
@@ -252,6 +252,23 @@ REST endpoint and interactive use.
...
+- **Interactive mode:** after ksqlDB Server restarts, it doesn't contact
+  {{ site.sr }} again to resolve schemas, because it has previously persisted
+  the information to the command topic.
+- **Non-interactive mode**: after ksqlDB Server restarts, it *does* contact
+  {{ site.sr }} again to resolve schemas. If schemas have changed, unexpected
+  behavior in your ksqlDB applications may occur.
+
+!!! important
+    If your ksqlDB applications use Avro and you run them in non-interactive
+    mode, ensure that the schemas don't change between ksqlDB Server restarts.
```
Comment (big-andy-coates) [inline docs-md/operate-and-deploy/installation/server-config/index.md]: OR they can choose to explicitly provide the schema.  This is safer if the schema may evolve.

This is the github issue that is tracking the feature for headless mode, if you want to reference it?

https://github.com/confluentinc/ksql/issues/1530

**Pull request merged**